drivers/lpsxxx/lpsxxx.c

```diff
@@ -0,0 +1,256 @@
+        DEBUG("[lpsxxx] init: cannot read WHO_AM_I register\n");
+        return -LPSXXX_ERR_I2C;
+    }
+
+    if (id != LPSXXX_WHO_AM_I) {
+        i2c_release(DEV_I2C);
+        DEBUG("[lpsxxx] init: not a valid device (got %02X, expected %02X)\n",
+              id, LPSXXX_WHO_AM_I);
+        return -LPSXXX_ERR_NODEV;
+    }
+
+    uint8_t tmp;
+
+#if MODULE_LPS22HB
+    if (i2c_read_reg(DEV_I2C, DEV_ADDR, LPSXXX_REG_CTRL_REG2, &tmp, 0) < 0) {
+        i2c_release(DEV_I2C);
+        DEBUG("[lpsxxx] init: cannot read LPSXXX_REG_CTRL_REG2 register\n");
+        return -LPSXXX_ERR_I2C;
+    }
+
+    /* Disable automatic increment of register address during byte access
+       (recommanded in datasheet (section 9.6 CTRL_REG2) */
+    tmp &= ~LPSXXX_CTRL_REG2_ID_ADD_INC;
+
+    DEBUG("[lpsxxx] init: update reg2, %02X\n", tmp);
+
+    if (i2c_write_reg(DEV_I2C, DEV_ADDR, LPSXXX_REG_CTRL_REG2, tmp, 0) < 0) {
+        i2c_release(DEV_I2C);
+        DEBUG("[lpsxxx] init: cannot write in CTRL_REG2 register\n");
+        return -LPSXXX_ERR_I2C;
+    }
+#endif
+
+    /* configure device, for simple operation only CTRL_REG1 needs to be touched */
+#if MODULE_LPS331AP
+    tmp = LPSXXX_CTRL_REG1_DBDU | LPSXXX_CTRL_REG1_PD |
+            (DEV_RATE << LPSXXX_CTRL_REG1_ODR_POS);
+#elif MODULE_LPS25HB
+    tmp = LPSXXX_CTRL_REG1_BDU | LPSXXX_CTRL_REG1_PD |
+            (DEV_RATE << LPSXXX_CTRL_REG1_ODR_POS);
+#elif MODULE_LPS22HB
+    tmp = LPSXXX_CTRL_REG1_EN_LPFP | /* Low-pass filter configuration: ODR/9 */
+            LPSXXX_CTRL_REG1_BDU | (DEV_RATE << LPSXXX_CTRL_REG1_ODR_POS);
+#endif
+
+    DEBUG("[lpsxxx] init: update reg1, value: %02X\n", tmp);
+
+    if (i2c_write_reg(DEV_I2C, DEV_ADDR, LPSXXX_REG_CTRL_REG1, tmp, 0) < 0) {
+        i2c_release(DEV_I2C);
+        DEBUG("[lpsxxx] init: cannot write in CTRL_REG1 register\n");
+        return -LPSXXX_ERR_I2C;
+    }
+
+    i2c_release(DEV_I2C);
+
+    DEBUG("[lpsxxx] initialization successful\n");
+    return LPSXXX_OK;
+}
+
+int lpsxxx_read_temp(const lpsxxx_t *dev, int16_t *temp)
+{
+    uint8_t tmp;
+    int16_t val = 0;
+    float res = TEMP_BASE;      /* reference value -> see datasheet */
+
+    i2c_acquire(DEV_I2C);
+    if (i2c_read_reg(DEV_I2C, DEV_ADDR, LPSXXX_REG_TEMP_OUT_L, &tmp, 0) < 0) {
+        i2c_release(DEV_I2C);
+        DEBUG("[lpsxxx] read_temp: cannot read TEMP_OUT_L register\n");
+        return -LPSXXX_ERR_I2C;
+    }
+    val |= tmp;
+
+    if (i2c_read_reg(DEV_I2C, DEV_ADDR, LPSXXX_REG_TEMP_OUT_H, &tmp, 0) < 0) {
+        i2c_release(DEV_I2C);
+        DEBUG("[lpsxxx] read_temp: cannot read TEMP_OUT_H register\n");
+        return -LPSXXX_ERR_I2C;
+    }
+    i2c_release(DEV_I2C);
+    val |= ((uint16_t)tmp << 8);
+
+    DEBUG("[lpsxxx] read_temp: raw data %08" PRIx32 "\n", (uint32_t)val);
+
+    /* compute actual temperature value in °C */
+    res += ((float)val) / TEMP_DIVIDER;
+
+    /* return temperature in c°C */
+    *temp = (int16_t)(res * 100);
```

Comment:
I think you can get rid of the float conversion & division:
```
(TEMP_BASE + (val / TEMP_DIVIDER)) * 100 == (TEMP_BASE * 100) + (val * 100 / TEMP_DIVIDER)
```

So this should be equivalent:

```
#define TEMP_BASE=4250U;
#define TEMP_DIVIDER=480U;

uint32_t res = TEMP_BASE;
uint32_t val = 0;
// ...
res += (val * 100) / TEMP_DIVIDER;
*temp = (int16_t)res;
```

(maybe use signed int).

Reply:
I tried but that doesn't work on lps331ap/lps25hb.